src/pandas_profiling/report/structure/variables/render_categorical.py

```diff
@@ -403,11 +403,13 @@ def render_categorical(config: Settings, summary: dict) -> dict:
 
     overview_items = []
 
-    if length:
+    # length isn't being computed for categorical in spark
+    if length and "max_length" in summary:
         length_table, length_histo = render_categorical_length(config, summary, varid)
         overview_items.append(length_table)
 
-    if characters:
+    # characters isn't being computed for categorical in spark
+    if characters and "category_alias_counts" in summary:
```

Comment:
will `category_alias_counts` be present for pandas?

Reply:
It is already and working for both.